File: DSA/ShlokMadan_590017851/Day_30/Question1.java
import java.util.Stack;

public class Question1 {

    public static String removeDuplicates(String s){

        Stack<Character> st=new Stack<>();

        for(int i=0;i<s.length();i++){

            char ch=s.charAt(i);

            if(!st.isEmpty() && st.peek()==ch)
                st.pop();
            else
                st.push(ch);
        }

        StringBuilder ans=new StringBuilder();

        while(!st.isEmpty())
            ans.append(st.pop());

        return ans.reverse().toString();
    }

    public static void main(String[] args){

        String s="abbaca";

        System.out.println(removeDuplicates(s));
    }
}

File: DSA/Pulkit_Singh_590022181/Day30/Question1.c
#include <stdio.h>
#include <string.h>

void removeDuplicates(char str[]) {

    char stack[1000];
    int top = -1;

    for (int i = 0; str[i] != '\0'; i++) {

        if (top != -1 && stack[top] == str[i]) {
            top--;
        }
        else {
            stack[++top] = str[i];
        }
    }

    stack[top + 1] = '\0';

    printf("Result: %s\n", stack);
}

int main() {

    char str[] = "abbaca";

    removeDuplicates(str);

    return 0;
}